### DISCUSSION TITLE
Extending a prismaObject in order to add fields

### DISCUSSION THREAD

**Discussion by cameronmakarchuk** [hayes/pothos#1323] [Q&A]
Hi,

I'm wondering what the best approach (if even possible) for creating a new type by extending a prismaObject to include additional fields not on the original prisma model?

For example, if we have a prisma model:
```
const User = Builder.prismaObject('UserModel', {
	fields: (t) => ({
		userID: t.exposeInt('id'),
		name: t.exposeString('name'),
```

I'd like to use "User" as the return type in a `queryType` field, to get the benefits of `...query`:

```
Builder.queryField(Constants.SchemaNames.PROMOTIONS_USER, (t) =>
    t.prismaField({
        type: 'User',
        resolve: (query, parent, args) => {
               const dbData = ...fetch from DB using ...query

               return {...dbData, ...additionalProperties}
   }
```

But the data going from the `queryType` resolver to the `User` type resolver is going to have additional properties not on the original `UserModel` prisma model, so it's giving an error.

Is there a way to interface a prismaObject and extend it with additional fields?

**Comment by hayes** [comment]
Unfortunately there isn't an existing pattern for that kind of extension.

The issue with this is that the prisma plugin needs to be able to fully re-load the data for its models to ensure that everything can be correctly resolved.

This basically means that you can't force your resolvers to return data that Pothos wouldn't also be able to load automatically, given the models ID.  

If you want to go down this route you have a couple of options:

* Don't use the prisma plugin for these models, using objectRefs or the dataloader plugin instead
* Shift the loading of the additionalProperties down into the individual fields
  * combining this with data loaders is probably what I'd do
  * You can create a data-loader backed helper like `getUserAdditionalData(user)` that you can use in your field resolvers

**Reply by cameronmakarchuk** [reply]
Thanks so much for the quick response. I think we'll take your suggestion with combining dataloader plugin and trying to shift some of the logic down to the individual fields instead.

Just out of curiosity, do you happen to have links to any examples doing something like this?

**Reply by hayes** [reply]
not that I can think, sorry
